src/Npgsql.NodaTime/IntervalHandler.cs

```diff
@@ -60,6 +63,31 @@ public override void Write(Period value, NpgsqlWriteBuffer buf, NpgsqlParameter?
             buf.WriteInt32(value.Years * 12 + value.Months); // months
         }
 
+        Duration INpgsqlSimpleTypeHandler<Duration>.Read(NpgsqlReadBuffer buf, int len, FieldDescription? fieldDescription)
+        {
+            var microsecondsInDay = buf.ReadInt64();
+            var days = buf.ReadInt32();
+            var totalMonths = buf.ReadInt32();
+
+            if (totalMonths != 0)
+                throw new NpgsqlException("Cannot read PostgreSQL interval with non-zero months to NodaTime Duration. Try reading as a NodaTime Period instead.");
+
+            return Duration.FromDays(days) + Duration.FromNanoseconds(microsecondsInDay * 1000);
+        }
+
+        public int ValidateAndGetLength(Duration value, NpgsqlParameter? parameter) => 16;
+
+        public void Write(Duration value, NpgsqlWriteBuffer buf, NpgsqlParameter? parameter)
+        {
+            var microsecondsInDay =
+                (((value.Hours * NodaConstants.MinutesPerHour + value.Minutes) * NodaConstants.SecondsPerMinute + value.Seconds) *
+                    MicrosecondsPerSecond + value.SubsecondNanoseconds / 1000); // Take the microseconds, discard the nanosecond remainder
```

Comment:
Or even make the constant local to this method. It's not used anywhere else. `1000` should be named too I think; otherwise, it's a bit magic.

Reply:
Made the constant local. I think in this context of calculating time units, the 1000 should be clear enough...